src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -799,7 +802,11 @@ def enable_zip_deploy(cmd, resource_group_name, name, src, timeout=None, slot=No
         if app_is_linux_webapp and track_status is not None and track_status:
             headers["x-ms-artifact-checksum"] = _compute_checksum(zip_content)
 
-        res = requests.post(zip_url, data=zip_content, headers=headers, verify=not should_disable_connection_verify())
+        if app_is_linux_webapp and not app_is_function_app:
```

Comment:
let's wrap this entire block in try except and fallback to original POST call incase of exception